libminifi/src/FlowController.cpp

```diff
@@ -205,13 +177,12 @@ int16_t FlowController::stop() {
      * -Stopping the schedulers doesn't actually quit the onTrigger functions of processors
      * -They only guarantee that the processors are not scheduled anymore
      * -After the threadpool is stopped we can make sure that processors don't need repos and controllers anymore */
-    if (this->root_) {
-      this->root_->drainConnections();
-    }
+    root_wrapper_.drainConnections();
     this->flow_file_repo_->stop();
     this->provenance_repo_->stop();
     // stop the ControllerServices
     disableAllControllerServices();
+    initialized_ = false;
```

Comment:
although `unload` is equivalent to calling `stop` and setting `initialized_`, conceptually the flow controller has the following state transitions: `empty` -(`load`)-> `initialized` -(`start`)-> `running` -(`stop`)-> `initialized` -(`unload`)-> `empty`

IMO we should either keep `unload` even if it can be inlined, or also remove `load`

Reply:
I think the `unload` was just complicating things as even in tests either `unload` or `stop` (or both of them together) was used  for the same functionality. `load` is a bit different as it also calls `stop` for the flow controller and reloads the flow controller if a new config is applied so I would keep that. I see what the original intent was, but it does not reflect the current functionality, so I would remove `unload` only.